apps/ejabberd/src/http_upload/http_upload_s3.erl

```diff
@@ -0,0 +1,144 @@
+%%==============================================================================
+%% Copyright 2016 Erlang Solutions Ltd.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%==============================================================================
+
+-module(http_upload_s3).
```

Comment:
I discussed it with the team before naming the module. There's a tradeoff in naming here - adding `mod_` associates the backend more tightly with the main module, on the other hand `mod_` suggests that it implements `gen_mod` behaviour. Since `mod_http_upload` backends (right now) have no business being `gen_mod`s, I've decided against adding `mod_` to `http_upload_s3` and instead associate modules more by putting them in the same directory.